packages/odata-service-inquirer/src/prompts/prompts.ts

```diff
@@ -0,0 +1,107 @@
+import { Severity } from '@sap-devx/yeoman-ui-types';
+import { withCondition, type YUIQuestion } from '@sap-ux/inquirer-common';
+import type { Answers, Question } from 'inquirer';
+import { t } from '../i18n';
+import type { OdataServiceAnswers, OdataServiceQuestion } from '../types';
+import {
+    DatasourceType,
+    promptNames,
+    type DatasourceTypePromptOptions,
+    type OdataServicePromptOptions
+} from '../types';
+import { getMetadataFileQuestion } from './datasources/metadata-file';
+import { getDatasourceTypeChoices } from './prompt-helpers';
+import { getLocalCapProjectPrompts } from './datasources/cap-project/questions';
+import LoggerHelper from './logger-helper';
+
+/**
+ * Get the prompts for the OData service inquirer.
+ *
+ * @param promptOptions
+ * @returns the prompts used to provide input for OData service generation
+ */
+export async function getQuestions(promptOptions?: OdataServicePromptOptions): Promise<OdataServiceQuestion[]> {
+    const questions: OdataServiceQuestion[] = [getDatasourceTypeQuestion(promptOptions?.datasourceType)];
+
+    // Add conditional questions depending on the selected source
+    questions.push(...(await getDatasourceTypeConditionalQuestions(promptOptions)));
+
+    return questions;
+}
+
+/**
+ * Get the datasource type question.
+ *
+ * @param options
+ * @returns the datasource type question
+ */
+function getDatasourceTypeQuestion(options?: DatasourceTypePromptOptions): YUIQuestion<OdataServiceAnswers> {
+    const choices = getDatasourceTypeChoices(options);
+    return {
+        type: 'list',
+        name: promptNames.datasourceType,
+        guiOptions: {
+            breadcrumb: true
+        },
+        default: options?.default ?? -1,
```

Comment:
Curious as to what -1 does?

Reply:
None selected so you get the message `Please choose...`